pkg/object/rawconfigtrafficcontroller/rawconfigtrafficcontroller.go

```diff
@@ -0,0 +1,211 @@
+/*
+ * Copyright (c) 2017, MegaEase
+ * All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package rawconfigtrafficcontroller
+
+import (
+	"fmt"
+
+	"github.com/megaease/easegress/pkg/logger"
+	"github.com/megaease/easegress/pkg/object/httppipeline"
+	"github.com/megaease/easegress/pkg/object/httpserver"
+	"github.com/megaease/easegress/pkg/object/trafficcontroller"
+	"github.com/megaease/easegress/pkg/supervisor"
+)
+
+const (
+	// Category is the category of RawConfigTrafficController.
+	Category = supervisor.CategorySystemController
+
+	// Kind is the kind of RawConfigTrafficController.
+	Kind = "RawConfigTrafficController"
+
+	DefaultNamespace = "default"
+)
+
+type (
+	// RawConfigTrafficController is a system controller to manage
+	// TrafficGate, Pipeline and their relationship.
+	RawConfigTrafficController struct {
+		super     *supervisor.Supervisor
+		superSpec *supervisor.Spec
+		spec      *Spec
+
+		watcher   *supervisor.ObjectEntityWatcher
+		tc        *trafficcontroller.TrafficController
+		namespace string
+		done      chan struct{}
+	}
+
+	// Spec describes RawConfigTrafficController.
+	Spec struct {
+	}
+
+	// Status is the status of RawConfigTrafficController.
+	Status = trafficcontroller.StatusOneSpace
```

Comment:
According to` easemonitormetrics.go` implementation, `StatusOneSpace` seems to only be used to let HTTPServer and HTTPPiepline's status records reporting to the same metric name. (SameSpace?) Proposal to add more comments about this status or change it to a more meaningful name.(StatusSameNamespace)

Reply:
Updated.